app/scripts/lib/transaction/util.ts

```diff
@@ -66,8 +84,45 @@ export async function addDappTransaction(
 export async function addTransaction(
   request: AddTransactionRequest,
 ): Promise<TransactionMeta> {
-  const { waitForSubmit } = request;
+  ///: BEGIN:ONLY_INCLUDE_IF(blockaid)
+  const {
+    transactionParams,
+    transactionOptions,
+    ppomController,
+    securityAlertsEnabled,
+    chainId,
+  } = request;
+
+  if (securityAlertsEnabled && SUPPORTED_CHAIN_IDS.includes(chainId)) {
+    let securityAlertResponse;
+
+    try {
+      const ppomRequest = {
+        method: 'eth_sendTransaction',
+        id: 'actionId' in transactionOptions ? transactionOptions.actionId : '',
+        origin: 'origin' in transactionOptions ? transactionOptions.origin : '',
```

Comment:
```suggestion
        id: transactionOptions.actionId ?? '',
        origin: transactionOptions.origin ?? '',
```
I believe this would also work

Reply:
The aim of that code is to remove the IDE error `Property 'actionId' does not exist on type '{}'.ts(2339)` the only way is to check existence before accessing the property.